packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -149,7 +152,7 @@ export class ModuleGraph {
     for (const imported of importedModules) {
       const dep =
         typeof imported === 'string'
-          ? await this.ensureEntryFromUrl(imported, ssr)
+          ? this.urlToModuleMap.get(imported)!
```

Comment:
Anything in `importedModules` has a module node at this point, thanks to the `importAnalysis` plugin.